src/nodes.coffee

```diff
@@ -502,7 +502,7 @@ exports.Root = class Root extends Base
 
   commentsAst: ->
     @allComments ?=
-      for commentToken in (@allCommentTokens ? [])
+      for commentToken in (@allCommentTokens ? []) when not commentToken.heregex
```

Comment:
Exclude heregex comments from the top-level AST comments

They shouldn't be included there since eg Prettier would want to print heregex comments explicitly/separately rather than having them get printed as part of printing the heregex itself

So currently heregex comments aren't explicitly represented in AST (they're just included as part of the raw source pattern string for heregex AST nodes). If we felt like they should get exposed as "parsed" comments, maybe we could give heregex AST nodes their own separate `comments` list (that Prettier would ignore)? But I'd want to think about that a bit more

Reply:
Comments within block regexes have been a tough case, since JavaScript itself doesn’t support comments within a regex literal. See https://coffeescript.org/#regexes: the comments there are just appended after the regex. Which isn’t ideal, and not very informative, but better (I think) than simply not outputting the comments at all. Can we do the AST equivalent of that?

Reply:
Ok I added support/tests for exposing a `comments` property on heregex AST nodes (interpolated/non-interpolated)

This involved attaching correct location data to heregex comments and then using the typical technique for passing them through the grammar so that they end up explicitly passed into `RegexLiteral`/`RegexWithInterpolations` nodes